templatehelper_test.go

```diff
@@ -0,0 +1,43 @@
+package termenv
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"text/template"
+)
+
+func TestTemplateFuncs(t *testing.T) {
+	tests := []struct {
+		name    string
+		profile Profile
+	}{
+		{"ascii", Ascii},
+		{"ansi", ANSI},
+		{"ansi256", ANSI256},
+		{"truecolor", TrueColor},
+	}
+	const templateFile = "./testdata/templatehelper.tpl"
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tpl, err := template.New("templatehelper.tpl").Funcs(TemplateFuncs(test.profile)).ParseFiles(templateFile)
+			if err != nil {
+				t.Fatalf("unexpected error parsing template: %v", err)
+			}
+			var buf bytes.Buffer
+			if err = tpl.Execute(&buf, nil); err != nil {
+				t.Fatalf("unexpected error executing template: %v", err)
+			}
+			actual := buf.Bytes()
+			filename := fmt.Sprintf("./testdata/templatehelper_%s.txt", test.name)
+			expected, err := ioutil.ReadFile(filename)
+			if err != nil {
+				t.Fatalf("unexpected error reading golden file %q: %v", filename, err)
+			}
+			if !bytes.Equal(buf.Bytes(), expected) {
+				t.Fatalf("template output does not match golden file.\n--- Expected ---\n%s\n--- Actual ---\n%s\n", string(expected), string(actual))
+			}
```

Comment:
I'm not the biggest fan of doing this kind of comparison and output in tests, but I didn't want to introduce a test-only dependency unless you're ok with it.  

A couple I had in mind to make these sorts of tests more readable/maintainable:
- https://github.com/sebdah/goldie for managing golden files (you can run the test with -update to update the files from the code output)
- https://github.com/stretchr/testify has test assertions with useful textual diffs.

Reply:
I agree, that's probably a sensible suggestion. I haven't looked into `goldie` yet, but I've used `testify` before.